src/geometry/Point.js

```diff
@@ -26,9 +26,10 @@ import {isArray, formatNum} from '../core/Util';
 
 export function Point(x, y, round) {
 	// @property x: Number; The `x` coordinate of the point
-	this.x = (round ? Math.round(x) : x);
+	this.x = (round ? Math.round(x) : parseFloat(x));

```

Comment:
Good!
But I worry about this case: `parseFloat("1x") === 1`.

Reply:
What should it do, when `"1x"` is passed? An error?

Reply:
Well, maybe just `NaN`, I am not sure.
See more general issue: https://github.com/Leaflet/Leaflet/issues/7128

Reply:
It looks like that Leaflet today don't validate inputs / values, maybe in future but I suggest that this should not be a part of this PR. 
In my eyes the value "1x" is invalid so why not parse it to "1"? Input is not a number as in the doku defined, so it can happens that an unexpected result comes out.

Reply:
> In my eyes the value "1x" is invalid so why not parse it to "1"?

Invalid values should be rejected, otherwise it may lead to hard-to-find bugs.

> but I suggest that this should not be a part of this PR.

Well, may be.

Reply:
> Invalid values should be rejected, otherwise it may lead to hard-to-find bugs.

Ok yeah this can be a problem

What do you think about:
```js
export function Point(x, y, round) {
        x = isNaN(x) ? x : parseFloat(x);
        y = isNaN(y) ? y : parseFloat(y);
	// @property x: Number; The `x` coordinate of the point
	this.x = (round ? Math.round(x) : x);
	// @property y: Number; The `y` coordinate of the point
	this.y = (round ? Math.round(y) : y);
}
```

Reply:
`new Point('five', 'six').add([0,0]) === {x: 'five0', y: 'six0'}`?

Reply:
Ok let it change to `x = isNaN(x) ? NaN : parseFloat(x);` then it always returns `NaN` also after called `add([0,0])`

Reply:
May be better to throw explicitly, like here:
https://github.com/Leaflet/Leaflet/blob/436430db4203a350601e002c8de6a41fae15a4bf/src/geo/LatLng.js#L30-L33

Reply:
Yes this sounds better, I updated it